enhancements/update/reduced_reboots.md

```diff
@@ -0,0 +1,328 @@
+status: provisional
+see-also:
+  - https://github.com/openshift/enhancements/pull/585
+  - "/enhancements/eus-mvp.md"
+
+---
+
+# Reduced Reboot Upgrades
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement is intended to reduce host reboots when upgrading across two or
+more OpenShift minor versions by enabling an N-2 version skew policy between all
+host components and cluster scoped resources.
+
+## Motivation
+
+While OpenShift is designed to minimize workload disruption and risk associated
+with rolling reboots there exist a class of customers and workloads where
+reboots remain a disruptive and time consuming activity. Additionally, with the
+introduction of Extended Update Support (EUS) a new upgrade pattern will emerge
+where clusters run 4.6 for a year or more then rapidly upgrade across multiple
+minor versions in a short period of time. Those customers wish to complete their
+upgrades in a condensed time frame and with as few reboots as possible, they do
+not intend to run each minor version for an extended period of time.
+
+### Goals
+
+- Define testing requirements for N-2 host to cluster resource version skew
+- Define version skew policies for host and cluster scoped resources
+- Reduce reboots in accordance with our new tested policies
+
+### Non-Goals
+
+- Exceeding upstream's documented Kubelet version skew policies
+
+## Proposal
+
+### User Stories
+
+#### Node - Improve Upstream Kubelet Version Skew Testing
+
+Kubernetes defines a version skew policy(https://kubernetes.io/docs/setup/release/version-skew-policy/#kubelet)
+which allows for kubelet N-2 to be compatible with kube-apiserver version N. At
+this point in time OpenShift is not comfortable with the level of testing upstream
+and the intersection of the specific features of OpenShift. We should work to
+define and implement upstream testing changes which give us an appropriate level
+of confidence that N-2 version skew issues would be identified in the community
+whenever possible.
+
+#### OTA - Implement Downstream Paused Worker Pool Upgrade Tests
+
+In parallel with efforts to revamp upstream version skew testing we must also
+implement downstream version skew testing which includes any additional tests
+required for OpenShift specific implementation details.
+
+We will achieve this by delivering upgrade jobs which pause the Worker MachineConfigPool
+then upgrade from 4.x to 4.x+1 to 4.x+2. We will run conformance tests from 4.x
+after upgrading to 4.x+1 in order to ensure that we continue to provide a baseline
+feature set, then again after upgrading to 4.x+2, finally after unpausing the
```

Comment:
I'm not clear on why we couldn't run the 4.x+1 suite on the cluster once the control-plane reached 4.x+1.  It seems unlikely to me that the test suite is tightly tied to kubelet-side features, and that when it is, it requires 4.x+1 _compute_ nodes.  The distinction isn't critical, because leaving a pool paused long enough to port workloads to new APIs is unwise, but running the tests that match the current control plane seems convenient if it works.

Reply:
Yeah, it's not a straight forward question to answer. We know that under our current upgrade process most operators have to tolerate some incomplete upgrade state without exploding because MCO updates nodes (kubelets) toward the very end of the upgrade, but dodging problems there isn't quite the same as running a full test suite. At the same time it's not truly a 4.x+1 cluster so if we do choose to run the tests we should not be surprised if there's some portion which leverage new features that fail.

I have a feeling that where we'll run into problems is when we upgrade the control plane to 4.x+2 but we still have kubelets at 4.x. This is where we'll likely run into operators which have always assumed a baseline feature set of 4.x+1.

Perhaps node team have an opinion here? @rphillips @ehashman @harche